### PR TITLE
Document header ordering caveats.

### PR DESCRIPTION
**Pull request by Lukasa**
Resolves #3096.

### CODE REVIEW

**Comment by sigmavirus24** [comment]
This looks great. Thanks @Lukasa

File: docs/user/advanced.rst [hunk 1]
```diff
@@ -899,6 +899,13 @@ Two excellent examples are `grequests`_ and `requests-futures`_.
 .. _`grequests`: https://github.com/kennethreitz/grequests
 .. _`requests-futures`: https://github.com/ross/requests-futures
 
+Header Ordering
+---------------
+
+In unusual circumstances you may want to provide headers in an ordered manner. If you pass an ``OrderedDict`` to the ``headers`` keyword argument, that will provide the headers with an ordering. *However*, the ordering of the default headers used by requests will be preferred, which means that if you override default headers in the ``headers`` keyword argument, they may appear out of order compared to other headers in that keyword argument.
```
Comment (kennethreitz) [inline docs/user/advanced.rst:905]: Requests is a proper noun and should always be capitalized.

**Comment by 0x3c3e** [comment]
@Lukasa can you supply details how to change headers in Session object? Tried `requests.Session.headers = headers`, but it doesn't work (no effect)

**Comment by nateprewitt** [comment]
@0x3c3e, please consult the documentation for Session. The code you’ve supplied is modifying the Session class, not an instance of Session which is likely your problem. Further questions about usage should directed towards StackOverflow as per our project policies. Thanks!

**Pull request merged**